code/modules/uplink/uplink_items.dm

```diff
@@ -286,610 +325,505 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
-	player_minimum = 25
-	restricted = TRUE
-	refund_path = /obj/item/guardiancreator/tech/choose/traitor
+	include_modes = list(/datum/game_mode/nuclear/clown_ops)
 
-/datum/uplink_item/dangerous/machinegun
+/datum/uplink_item/munitions/machinegun
 	name = "L6 Squad Automatic Weapon"
-	desc = "A fully-loaded Aussec Armoury belt-fed machine gun. \
-			This deadly weapon has a massive 50-round magazine of devastating 1.95x129mm ammunition."
+	desc = "A heavy duty Aussec Armoury belt-fed machine gun that carries a massive 100-round magazine of armor piercing 5.56�45mm ammunition that \
+			is fired in 5 round bursts. Requires two hands to fire. The L6 is ideal for suppressive fire and support roles."
 	item = /obj/item/gun/ballistic/automatic/l6_saw
 	cost = 18
 	surplus = 0
 	include_modes = list(/datum/game_mode/nuclear)
 
-/datum/uplink_item/dangerous/carbine
-	name = "M-90gl Carbine"
-	desc = "A fully-loaded, specialized three-round burst carbine that fires 5.56mm ammunition from a 30 round magazine \
-			with a toggleable 40mm underbarrel grenade launcher."
-	item = /obj/item/gun/ballistic/automatic/m90
-	cost = 18
-	surplus = 50
-	include_modes = list(/datum/game_mode/nuclear)
-
-/datum/uplink_item/dangerous/powerfist
-	name = "Power Fist"
-	desc = "The power-fist is a metal gauntlet with a built-in piston-ram powered by an external gas supply.\
-		 Upon hitting a target, the piston-ram will extend forward to make contact for some serious damage. \
-		 Using a wrench on the piston valve will allow you to tweak the amount of gas used per punch to \
-		 deal extra damage and hit targets further. Use a screwdriver to take out any attached tanks."
-	item = /obj/item/melee/powerfist
-	cost = 8
-
-/datum/uplink_item/dangerous/sniper
-	name = "Sniper Rifle"
-	desc = "Ranged fury, Syndicate style. Guaranteed to cause shock and awe or your TC back!"
+/datum/uplink_item/munitions/sniper
+	name = "Anti-Material Sniper Rifle"
+	desc = "A long-range, scoped, specialist rifle that fires devastating .50 caliber rounds from a 5 round magazine. \
+			Incredible at neutralizing nearly any kind of individual target, but absolutely terrible at close range, or against multiple opponents."
 	item = /obj/item/gun/ballistic/automatic/sniper_rifle/syndicate
 	cost = 16
 	surplus = 25
 	include_modes = list(/datum/game_mode/nuclear)
 
-/datum/uplink_item/dangerous/bolt_action
-	name = "Surplus Rifle"
-	desc = "A horribly outdated bolt action weapon. You've got to be desperate to use this."
-	item = /obj/item/gun/ballistic/shotgun/boltaction
-	cost = 2
+/datum/uplink_item/munitions/ion
+	name = "Ion Rifle"
+	desc = "A man-portable anti-electronics weapon designed to disrupt electrical systems and mechanical targets."
+	item = /obj/item/gun/energy/ionrifle
+	cost = 12
+	include_modes = list(/datum/game_mode/nuclear)
+
+
+//CURRENTLY UNABLE TO LOAD ROCKETS INTO LAUNCHER. UNCOMMENT ONCE FIXED
```

Comment:
I should have fixed this like a month ago. I even tested it locally and it was fine. Have you tried using the rocket launcher as of recently with an up to date branch?

Reply:
And yes, I must have fouled it up right after the test then.
I'll go on and fix it for real as fast as this ol' computer can go.